rfc/0009-expressions/README.md

```diff
@@ -0,0 +1,94 @@
+
+For future features (like value transformations), we also need boolean expressions that allow for numeric comparisons, e.g. `3 > 5` should resolve to `false`.
+
+## Explanation
+
+### Operators
+
+The following operators are in descending order by their precedence:
+- `()` for grouping a sub-expression
+- unary operators (see below)
+- binary operators (see below)
+
+#### Unary Prefix Operatory
+Binary prefix operators follow the pattern `op a`. Brackets can be used for grouping larger sub-expressions.
+- `floor`, `ceil`, `round` for conversion of decimal numbers to integers.
+- `not` for unary inversion of boolean values
+
+#### Binary Operators
+Binary infix operators follow the infix pattern `a op b`.
+- `pow`, `root` for power and root calculation. E.g., `2 pow 3` is 2³, `5 root 2` is `sqrt(5).
+- `*`, `/`, `%` for multiplication, division, and remainder
+- `+`, `-` for addition and subtraction
+- `<`, `<=`, `>`, `>=` for relational operators comparing numbers
+- `==`, `!=` for equality operators comparing numbers and booleans
+- `and` for a logical AND on booleans
+- `xor` for a logical XOR on booleans
+- `or` for a logical OR on booleans
+
+
+### Handling of booleans
+
+The expression resolving of boolean does follow the common pattern. 
+When operators of similar precedence don't determine the order, we evaluate from left to right.
+
+
+### Handling of numbers
+
+Division and remainder operators **always** produce `decimal` values. To convert to an integer value, the operators `floor`, `ceil`, `round` have to be used.
+
+Division by zero should produce an `INVALID` value in the future. For now, it will just throw an error. 
+
+### Interplay of booleans and numbers
+
+We infer the datatype based on the operator. The resulting value type of each operator unambiguous. 
+
+The operands `==` and `!=` are the only ones that allow operands of different types. Other operators are unambiguous. Comparing the equality of two operands of different value types always resolves to `false`.  
+
+
+## Drawbacks
+
+- No strings supported yet.
+- Some operators are functions in other languages which might be unintuitive.
+- Always converting to decimals on division might lead to misleading results if the programmer is not aware.
+
+## Alternatives
+
+- Use an established scripting language instead (e.g. Lua script).
+- Automatically convert to integer values based on some criteria.
+- Make `pow` and `root` prefix operators, e.g., `root(5, 2)`.
+- Require grouping brackets on unary operators.
+- Equality check could automatically convert value types into each other if possible.
+
+## Possible Future Changes/Enhancements
+
+- Incorporate operators for `text` values (append, split, get length, ...).
+- Access input data (e.g., sheet cells) to implement value transformations (#213).
```

Comment:
Two more points:

- Boolean expressions could be used to describe custom constraints
- Depending on our future design decisions, value transformations may relate more to value representation concepts (e.g. value readers / writers) rather than blocks.